README.md

```diff
@@ -1105,6 +1110,59 @@ However, the following rules of thumb are useful to keep in mind:
   Alternatively, the entire OrientDB class and all edges that may point to it may be omitted
   entirely from the GraphQL schema.
 
+## Query cost estimation
+
+Due to GraphQL's easy-to-use nature, queries that use a lot of memory and/or have a long execution
+time can happen often. For better user-friendliness, it's worthwhile to estimate a query's cost (in
+terms of time taken and memory used) using statistics without executing the query on the database.
+Since the time and memory required for query execution varies depending on the underlying database,
+as a rough proxy, the cost estimator estimates the number of result rows for the given query i.e.
+the query's cardinality.
+
+These estimates can be used to improve user-friendliness for query execution.
+For example, the estimates can be used to warn users about a query that asks for too many results.
+
+### Usage of query cost estimation
+
+To get estimates for the number of result rows for a given query and statistics object, use the
+`estimate_query_result_cardinality()` function in
+`graphql_compiler/cost_estimation/cardinality_estimator.py`.
+
+### Statistics
+
+Before estimates can be generated, you need to provide statistics about the graph data. The
+applicable classes are in `graphql_compiler/cost_estimation/statistics.py`. Currently, two classes
+are implemented:
+- `Statistics`, an abstract base class that describes the interface the cost estimator expects for
```

Comment:
Since LocalStatistics is the only implementation right now let's remove the Statistics base class mention from the README.md.

Reply:
I'm not so sure. Creating your own implementation of Statistics requires implementing three functions, and LocalStatistics is not what you need.